### ISSUE TITLE
FMT_USE_NONTYPE_TEMPLATE_ARGS detection is wrong on trunk

### ISSUE THREAD

**Issue by timsong-cpp** [fmtlib/fmt#3935]
With trunk GCC and `-std=c++2c `:

```cpp
#include <fmt/compile.h>
using namespace fmt::literals;

int main() {
    fmt::print(""_cf);
}
```

does not compile with trunk fmt but does with 10.2.1:

https://gcc.godbolt.org/z/4WsvGP6bs

https://github.com/fmtlib/fmt/blob/aa52eb765dec0eb67fab4b035a4a05b82346c8f5/include/fmt/base.h#L143-L144 looks suspicious.

**Comment by vitaut** [comment]
Good catch, thanks! Looks like I accidentally inverted this check when cleaning it up.

**Comment by vitaut** [comment]
Fixed in #3937 (thanks @Arghnews).
